src/ragas/metrics/base.py

```diff
@@ -723,3 +727,619 @@ class ModeMetric(t.Protocol):
 
 
 ensembler = Ensember()
+
+
+@dataclass
+class SimpleBaseMetric(ABC):
+    """Base class for simple metrics that return MetricResult objects."""
+
+    name: str
+
+    @abstractmethod
+    def score(self, **kwargs) -> "MetricResult":
+        pass
+
+    @abstractmethod
+    async def ascore(self, **kwargs) -> "MetricResult":
+        pass
+
+    def batch_score(
+        self,
+        inputs: t.List[t.Dict[str, t.Any]],
+    ) -> t.List["MetricResult"]:
+        return [self.score(**input_dict) for input_dict in inputs]
+
+    async def abatch_score(
+        self,
+        inputs: t.List[t.Dict[str, t.Any]],
+    ) -> t.List["MetricResult"]:
+        async_tasks = []
+        for input_dict in inputs:
+            # Process input asynchronously
+            async_tasks.append(self.ascore(**input_dict))
+
+        # Run all tasks concurrently and return results
+        return await asyncio.gather(*async_tasks)
+
+
+def create_auto_response_model(name: str, **fields):
+    """Create a response model and mark it as auto-generated by Ragas.
+
+    This function creates a Pydantic model using create_model and marks it
+    with a special attribute to indicate it was auto-generated. This allows
+    the save() method to distinguish between auto-generated models (which
+    are recreated on load) and custom user models.
+
+    Parameters:
+    -----------
+    name : str
+        Name for the model class
+    **fields
+        Field definitions in create_model format
+
+    Returns:
+    --------
+    Type[BaseModel]
+        Pydantic model class marked as auto-generated
+    """
+    from pydantic import create_model
+
+    model = create_model(name, **fields)
+    model.__ragas_auto_generated__ = True
+    return model
+
+
+@dataclass(repr=False)
+class SimpleLLMMetric(SimpleBaseMetric):
+    """LLM-based metric that uses prompts to generate structured responses."""
+
+    prompt: t.Optional[t.Union[str, "Prompt"]] = None
+    _response_model: t.Type["BaseModel"] = field(init=False)
+
+    def __post_init__(self):
```

Comment:
Some metrics can be created without prompts but will fail during save.

Either we can make prompts required here or allow None in serialization..